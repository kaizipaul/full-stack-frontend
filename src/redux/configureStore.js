import { configureStore, combineReducers } from '@reduxjs/toolkit';
// import { carsReducer } from './cars/carsView';
import reservationReducer from './reservation/reserve';
import createCarReducer from './newCar/car';
import deleteCarReducer from './deleteCar/deleteCar';
import { authReducer } from './auth/auth';

const token = localStorage.getItem('token');
const initialState = {
  auth: {
    token: token || null,
    isAuthenticated: !!token,
  },
};

const rootReducer = combineReducers({
  // cars: carsReducer,
  newcar: createCarReducer,
  auth: authReducer,
  delete: deleteCarReducer,
  reservations: reservationReducer,
});

export default configureStore({
  reducer: rootReducer,
  preloadedState: initialState,
});
