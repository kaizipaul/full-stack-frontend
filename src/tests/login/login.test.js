import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from '../../redux/configureStore';
import LoginForm from '../../pages/login/Login';

jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve({ data: 'mock data' })),
}));

describe('Login Form', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={configureStore}>
        <BrowserRouter>
          <LoginForm />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
