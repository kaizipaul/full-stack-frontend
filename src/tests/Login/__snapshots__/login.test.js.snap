// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-cont"
      >
        <form
          class="login-form"
        >
          <h1
            class="sidebar-title-signup pt-3 ml-5"
          >
            WheelWizard
          </h1>
          <div
            class="mb-3"
          >
            <input
              class="form-control"
              id="formBasicEmail"
              placeholder="Enter email"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <input
              class="form-control"
              id="formBasicPassword"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-primary"
            type="submit"
          >
            Login
          </button>
          <div
            class="d-flex account"
          >
            <p>
              No account yet? Click here to Sign Up
            </p>
            <button
              class="btn plain-btn btn btn-white"
              type="button"
            >
              Sign up
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-cont"
    >
      <form
        class="login-form"
      >
        <h1
          class="sidebar-title-signup pt-3 ml-5"
        >
          WheelWizard
        </h1>
        <div
          class="mb-3"
        >
          <input
            class="form-control"
            id="formBasicEmail"
            placeholder="Enter email"
            required=""
            type="email"
            value=""
          />
        </div>
        <div
          class="mb-3"
        >
          <input
            class="form-control"
            id="formBasicPassword"
            placeholder="Password"
            required=""
            type="password"
            value=""
          />
        </div>
        <button
          class="btn btn-primary"
          type="submit"
        >
          Login
        </button>
        <div
          class="d-flex account"
        >
          <p>
            No account yet? Click here to Sign Up
          </p>
          <button
            class="btn plain-btn btn btn-white"
            type="button"
          >
            Sign up
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
